src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -57,7 +55,9 @@ public class BodyModelBinder : BindingSourceModelBinder
                 var unsupportedContentType = Resources.FormatUnsupportedContentType(
                     bindingContext.OperationBindingContext.HttpContext.Request.ContentType);
                 bindingContext.ModelState.AddModelError(bindingContext.ModelName, unsupportedContentType);
-                return new ModelBindingResult(null, bindingContext.ModelName, isModelSet: false);
+
+                // Base class will create a ModelBindingResult that exits current ModelBinding loop.
+                return null;
```

Comment:
I don't agree at all that these are special cases. The contract of a model binder is that you return a `ModelBinderResult` if you *handled it* and `null` if you did not.  

Handling it might mean that you failed to bind, and we provide a way to indicate to the world (`isModelSet = false`). Doing things differently in two places just requires more explanation and comments.